src/shredder_flow/view/FunctionGeneratorInvoker.java

```diff
@@ -32,34 +46,278 @@ public void invokeGeneration() {
-		final JButton button = new JButton();
+		button = new JButton();
 		button.setAction(action);
 		shrinkPanel.add(button);
 		button.setText(caption);
 	}
 
+	private void buildFunction() {
+
+		handwrittenFunctionPannel.removeAll();
+		handwrittenFunctionPannelSum.removeAll();
+
+		handwrittenFunctionPannel.add(new JLabel("f(x,y)="));
+		alphaTextField = new JTextField("1.0");
+		handwrittenFunctionPannel.add(alphaTextField);
+
+		handwrittenFunctionPannel.add(new JLabel("*x^"));
+		xExponentTextField = new JTextField("1.0");
+		handwrittenFunctionPannel.add(xExponentTextField);
+
+		handwrittenFunctionPannel.add(new JLabel("*y^"));
+		yExponentTextField = new JTextField("1.0");
+		handwrittenFunctionPannel.add(yExponentTextField);
+
+		handwrittenFunctionPannel.add(new JLabel("+"));
+		constantTextField = new JTextField("1.0");
+		handwrittenFunctionPannel.add(constantTextField);
+
+		addButton(new BuildFunctionAction(), "Apply");
+	}
+
+	private void buildFunctionWithSum() {
+
+		handwrittenFunctionPannel.removeAll();
+		handwrittenFunctionPannelSum.removeAll();
+
+		handwrittenFunctionPannelSum.add(new JLabel("f(x,y)="));
+		alphaTextField = new JTextField("1.0");
+		handwrittenFunctionPannelSum.add(alphaTextField);
+
+		handwrittenFunctionPannelSum.add(new JLabel("*x^"));
+		xExponentTextField = new JTextField("1.0");
+		handwrittenFunctionPannelSum.add(xExponentTextField);
+
+		handwrittenFunctionPannelSum.add(new JLabel("+"));
+		betaTextField = new JTextField("1.0");
+		handwrittenFunctionPannelSum.add(betaTextField);
+
+		handwrittenFunctionPannelSum.add(new JLabel("*y^"));
+		yExponentTextField = new JTextField("1.0");
+		handwrittenFunctionPannelSum.add(yExponentTextField);
+
+		handwrittenFunctionPannelSum.add(new JLabel("+"));
+		constantTextField = new JTextField("1.0");
+		handwrittenFunctionPannelSum.add(constantTextField);
+
+		addButton(new BuildFunctionSumAction(), "Apply");
+	}
+
+	private void performer() {
+		double alpha = 1;
+		double xUp = 1;
+		double yUp = 1;
+		double constant = 1;
+		try {
+			alpha = Double.parseDouble(alphaTextField.getText());
+		} catch (Exception e2) {
+			System.out
+					.println("WARNING: Could not convert given alpha to double.");
+		}
+		try {
+			xUp = Double.parseDouble(xExponentTextField.getText());
+		} catch (Exception e2) {
+			System.out
+					.println("WARNING: Could not convert given xUp to double.");
+		}
+		try {
+			yUp = Double.parseDouble(yExponentTextField.getText());
+		} catch (Exception e2) {
+			System.out
+					.println("WARNING: Could not convert given yUp to double.");
+		}
+		try {
+			constant = Double.parseDouble(constantTextField.getText());
+		} catch (Exception e2) {
+			System.out
+					.println("WARNING: Could not convert given constant to double.");
+		}
+		generator.generateHandwrittenFuction(alpha, xUp, yUp, constant);
+	}
+
+	private void performerSum() {
+		double alpha = 1;
+		double xUp = 1;
```

Comment:
`xUp` and `yUp` is actually `xExponent` and `yExponent`. Please rename (alt + shift + R).
Also change the  names that the user sees.